src/firefly/js/core/AppDataReducers.js

```diff
@@ -43,8 +43,8 @@ export function appDataReducer(state, action={}) {
             return updateSet(state, ['connections'], action.payload);
 
         case AppDataCntlr.APP_OPTIONS :
-            return updateSet(state, ['appOptions'], action.payload.appOptions);
-
+            const {overwrite,appOptions}= action.payload;
+            return overwrite ? {...state, appOptions} : {...state, appOptions:{...state.appOptions, ...appOptions}} ;
```

Comment:
It was a 'set/replace' operation.  You want to add a shallow 'merge' option?  I don't think a shallow merge is good enough since we have options with more than 1 level deep.

Reply:
OK, we should talk then because it was overwriting.